File: howtodobiotech/src/main/java/cz/utb/fai/howtodobiotech/services/users/AccountService.java
package cz.utb.fai.howtodobiotech.services.users;

import cz.utb.fai.howtodobiotech.models.users.Account;
import cz.utb.fai.howtodobiotech.repositories.users.AccountRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Service
public class AccountService {

    AccountRepository accountRepository;

    @Autowired
    public AccountService(AccountRepository accountRepository) {
        this.accountRepository = accountRepository;
    }

    public Optional<Account> getAccountById(Integer id) {
        return accountRepository.findById(id);
    }

    public List<Account> getAllAccounts() {
        List<Account> accounts = new ArrayList<Account>();
        accountRepository.findAll().forEach(accounts::add);

        return accounts;
    }

    public Account addAccount(Account account) {
        return accountRepository.save(account);
    }

    public Account updateAccount(Account account) {
        return accountRepository.save(account);
    }

    public void deleteAccountById(Integer id) {
        accountRepository.deleteById(id);
    }

    public  Optional<Account> findByUsername(String username){
        return accountRepository.findByUsername(username);

    }

    public  Boolean existsByUsername(String username){
        return accountRepository.existsByUsername(username);
    }

    public  Boolean existsByEmail(String email){
        return accountRepository.existsByEmail(email);
    };

}

File: howtodobiotech_fe/src/service/Account.service.js
import http from "../http-common";

class AccountService {
  getAllAccounts() {
    return http.get("/accounts");
  }

  getAccountById(id) {
    return http.get(`${"/accounts"}/${id}`);
  }

  createAccount(account) {
    return http.post("/accounts", account);
  }

  updateAccount(id, account) {
    return http.put(`${"/accounts"}/${id}`, account);
  }

  deleteAccount(id) {
    return http.delete(`${"/accounts"}/${id}`);
  }
}

const accountService = new AccountService();
export default accountService;
